Source/Core/DolphinQt/Config/Graphics/GeneralWidget.cpp

```diff
@@ -203,6 +203,8 @@ void GeneralWidget::SaveSettings()
         return;
       }
     }
+
+    Config::SetBase(Config::MAIN_GFX_BACKEND, current_backend);
```

Comment:
This looks sketchy to me. You're setting the Base here regardless of if the current backend difference is termporary (game config) or global. Yes, it seems like the only way to get into this function right now is by the user changing the dropdown manually, but still, ugly. This would also show the warning message for the Software Renderer for gameini-initiated changes if they got into this function.

May I suggest using `Config::GetActiveLayerForConfig(Config::MAIN_GFX_BACKEND) == Config::LayerType::Base` somewhere here to improve this?

Reply:
Though, even if you checked for that, this would trigger on the switchback from gameini-overridden to global again, wouldn't it? Also not great. Maybe I'm overthinking this...

Reply:
For what it's worth, actually testing it, it seems this behaves correctly as-is, so I am probably overthinknig this.

Reply:
I made some changes to hopefully address your feedback

Reply:
Yes, I think the current version makes more sense.